### PR TITLE
Sb clang support json arguments or command441

### PR DESCRIPTION
**Pull request by stevebillings**
# Pull Request template

**Link to github issue (if applicable):**

*

**If nothing above, what is your reason for this pull request:**

* Previously we expected the command field to be present in compile_commands.json. Turns out, either commands or arguments is required; commands is not required. So I added the arguments field to the data object, and a wrapper class that will provide whichever is present when asked for it.

**Changes proposed in this pull request:**

*
*
*

### CODE REVIEW

File: hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/clang/CompileCommandWrapper.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.blackducksoftware.integration.hub.detect.bomtool.clang;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.synopsys.integration.util.Stringable;
+
+public class CompileCommandWrapper extends Stringable {
```
Comment (taikuukaits) [inline hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/clang/CompileCommandWrapper.java]: I'm not sure I see the need for a wrapper. The getter methods on the Compile Command could do this. Or I'd prefer a word other than wrapper or this class. Something to consider.

File: hub-detect/src/test/groovy/com/blackducksoftware/integration/hub/detect/bomtool/clang/ClangExtractorTest.java [hunk 2]
```diff
@@ -189,4 +192,92 @@ public void testMultipleCommandsDependenciesPackages() throws IOException, Execu
         }
     }
 
+    @Test
+    public void testJsonWithArgumentsNotCommand() throws IOException, ExecutableRunnerException {
```
Comment (taikuukaits) [inline hub-detect/src/test/groovy/com/blackducksoftware/integration/hub/detect/bomtool/clang/ClangExtractorTest.java]: The size of the test is a concern.

File: hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/clang/CompileCommandWrapper.java (hunk 1 above)
Comment (stevebillings) [reply hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/clang/CompileCommandWrapper.java]: Here was my dilemma on that approach: The argument for public fields on JSON data objects was that that's how Gson treated them, so putting getters/setters in those classes mislead readers of the code into thinking they would be executed, when in fact, Gson goes around them. To stay true to that, I figured I'd better leave CompileCommand as-is (other than adding the field).

I agree on the name. Will try to think of a better one.

File: hub-detect/src/test/groovy/com/blackducksoftware/integration/hub/detect/bomtool/clang/ClangExtractorTest.java (hunk 2 above)
Comment (stevebillings) [reply hub-detect/src/test/groovy/com/blackducksoftware/integration/hub/detect/bomtool/clang/ClangExtractorTest.java]: Yes, that test has a lot of code in common with the previous one. Definitely a potential for refactoring.

**Pull request merged**